src/gordon_gcp/plugins/service/event_consumer.py

```diff
@@ -304,6 +315,21 @@ def _get_and_validate_pubsub_msg_schema(self, pubsub_msg_data):
             self._logger, {'msg_id': msg_id})
         msg_logger.info('Received new message.')
 
+        # The timestamp is in a weird format; see
+        # https://pythonhosted.org/gax-google-logging-v2/google.protobuf.timestamp_pb2.html
+        # It divides a time with nanosecond precision into an integer part and
+        # a nanoseconds part.  To turn them into a regular float, we need to
+        # divide the nanos by a billion (nano = billionth) and add.
+        # E.g.: 1234 seconds and 123456789 nanos -> 1234.123456789
```

Comment:
Could we just do seconds? It would simplify things a bit and I'm not sure we need/have more accuracy than that.

Reply:
If we do that we'd have to truncate time.time().  And we could still introduce weird edge cases in  which something past the limit by a fraction of a second doesn't get dropped or vice versa.

Reply:
Yeah, but my question was more, is that problem? If we miss acking or not acking a message because it arrived within a fraction of second delta from our arbitrarily defined threshold which is likely on the scale of minutes.

Reply:
I mean, it seems like potentially confusing behavior that only saves us like 2 lines of code, vs. doing what I'd consider the "correct" thing.  If everyone hates it i can change it.

Reply:
Ok, so I spoke to @descent3me and we're ok with removing the seconds and adding a disclaimer about slight inaccuracy.  BUT.  It turns out that the behavior exhibited by the latest release of google-cloud-pubsub is broken.  The latest code in github has a fix, but evidently they commit everything to master and only release new packages sporadically, so it wasn't clear why the code I get when I download didn't match github...

TLDR: I'm going to pin the library to the commit with the fix and change all this back to using datetimes, which also removes the nanos.